flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/deployment/AbstractDeploymentObserver.java

```diff
@@ -252,6 +257,56 @@ private void onMissingDeployment(FlinkDeployment deployment) {
                 EventUtils.Component.JobManagerDeployment);
     }
 
+    /**
+     * Checks a deployment that is currently in the UPGRADING state whether it was already deployed
+     * but we simply miss the status information. After comparing the target resource generation
+     * with the one from the possible deployment if they match we update the status to the already
+     * DEPLOYED state.
+     *
+     * @param flinkDep Flink resource to check.
+     * @param context Context for reconciliation.
+     */
+    private void checkIfAlreadyUpgraded(FlinkDeployment flinkDep, Context context) {
+        Optional<Deployment> depOpt = context.getSecondaryResource(Deployment.class);
+        var status = flinkDep.getStatus();
+        depOpt.ifPresent(
+                deployment -> {
+                    Map<String, String> annotations = deployment.getMetadata().getAnnotations();
+                    if (annotations == null) {
+                        return;
+                    }
+                    Long deployedGeneration =
+                            Optional.ofNullable(annotations.get(FlinkUtils.CR_GENERATION_LABEL))
+                                    .map(Long::valueOf)
+                                    .orElse(-1L);
+
+                    // For first deployments we get the generation from the metadata directly
```

Comment:
nit: after moving the logic to a utility method, this comment is seemingly redundant

Reply:
yes, forgot this one